packages/plugin-core/src/components/views/PreviewViewFactory.ts

```diff
@@ -15,13 +15,19 @@ export class PreviewPanelFactory {
   /**
    * Get a usable PreviewProxy for showing the preview
    */
-  public static create(extension: IDendronExtension): PreviewProxy {
+  public static create(
+    extension: IDendronExtension,
+    linkHander?: IPreviewLinkHandler
+  ): PreviewProxy {
     // Simple singleton implementation, since we only want one preview panel at
     // any given time.
-    if (!PreviewPanelFactory._preview) {
+
+    // if preview panel doesn't exist yet, create a new one.
+    // if a custom link handler is provided, use that to create a new preview panel.
+    if (!PreviewPanelFactory._preview || linkHander) {
```

Comment:
I don't think it's too correct to have a singleton implemented this way.  Can we remove the `|| linkHandler` part?  I think it should still function fine since the `PreviewLinkHandler` implementation only acts on that one preview ID.

Reply:
merging `TutorialPreviewLinkHandler` and `PreviewLinkHandler` logic for now as per sync discussion